do away with the favicon144.png image

peterbe opened the pull request:
Part of #3611

Simplify simplify simplify. 
I ran a build using my trusted https://realfavicongenerator.net/ and looked at what meta tags they recommend, and it's this:

```html
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">
```

I'm going to ignore the last 3 because they're out of context. 
And I'm going to favor the `shortcut icon` (to use our `/favicon.ico`)

This removes those odd favicon images with the teal/blue background in and just keep the ones with a white dino on a black background. 

I also ran this new image through Optimage so now it's only 1,290 bytes.

schalkneethling commented on client/public/index.html:
```diff
@@ -4,31 +4,27 @@
     <link
-      rel="apple-touch-icon-precomposed"
-      sizes="144x144"
-      href="%PUBLIC_URL%/favicon144.png"
+      rel="shortcut icon"
+      href="%PUBLIC_URL%/favicon.ico"
+      type="image/x-icon"
     />
```
I don’t think we even need to specify this. If the browser does not support all the other formats we specify below, it will eventually go look at the root of the website for a `favicon.ico`

Ryuno-Ki replied:
In https://github.com/mdn/yari/pull/3621#pullrequestreview-644934488 it was recommended to drop the `shortcut` value for `rel`, too.

peterbe replied:
Lemme merge in `main` once https://github.com/mdn/yari/pull/3621 lands. 
This PR is supposed to be about removing the `favicon144.png` image.

schalkneethling commented on client/public/index.html:
```diff
@@ -4,31 +4,27 @@
+      type="image/png"
+      sizes="16x16"
+      href="%PUBLIC_URL%/favicon-16x16.png"
     />
```
I believe we also need:

```html
<!-- For Chrome on Android -->
<link rel="icon" href="/favicon-192x192.png" sizes="192x192" type="image/png" />
```

https://realfavicongenerator.net/blog/android-chrome-and-its-favicon/

schalkneethling reviewed (commented):
Two suggestions, but looks good. Thanks @peterbe

peterbe commented:
Ok. Nice clean-up now. 

- Added the 192x192 favicon too as per that blog post recommendation
- Rewrote the tests (used the home page instead of the `Web/Foo` page) and test more things
- Renamed the existing files (e.g. `android-chrome-512x512.png` becomes `favicon-512x512.png`)

peterbe commented:
@myakura Can you take a look at this one too please?

schalkneethling reviewed (approved):
w00t! Thanks, @peterbe!

myakura commented on client/public/index.html:
```diff
@@ -11,31 +11,28 @@
+      rel="icon"
+      type="image/png"
+      sizes="192x192"
+      href="%PUBLIC_URL%/favicon-192x192.png"
     />
```
Is it meant for Android devices? If so, this is unnecessary. Both Chrome and Firefox use apple-touch-icon or pick ones defined in manifest.json.

schalkneethling replied:
Technically yes but, https://realfavicongenerator.net/blog/android-chrome-and-its-favicon/ - Perhaps this is no longer an issue though? Would be good to know.

schalkneethling replied:
It looks like HTML5 boilerplate has literally scaled down to just two lines:
https://github.com/h5bp/html5-boilerplate/blob/master/src/index.html#L15

And then the 192x192 is specified in the webmanifest, with the `.ico` at the root so, we can follow suit?

myakura replied:
> Technically yes but, https://realfavicongenerator.net/blog/android-chrome-and-its-favicon/ - Perhaps this is no longer an issue though? Would be good to know.

That post is from 2014 so there should be changes made to browsers... I put the HTML up on https://valley-rainbow-gem.glitch.me/trimmed.html and both Chrome on Android and Fenix utilize the 180 file.

> It looks like HTML5 boilerplate has literally scaled down to just two lines:
> https://github.com/h5bp/html5-boilerplate/blob/master/src/index.html#L15
> 
> And then the 192x192 is specified in the webmanifest, with the `.ico` at the root so, we can follow suit?

so the simplest setup is

* a favicon.ico at root directory
* an apple-touch-icon in the link tag
* (a web app manifest file If you want some PWA experience)

but as per https://github.com/mdn/yari/issues/3611#issuecomment-825847727 it appears you cannot set longer cache TTL to the `/favicon.ico` URL. that's why there's a smaller 48x48 PNG file that has longer cache lifetime here.

schalkneethling replied:
So @myakura so, what your test indicates is that we can essentially go with:

1. favicon.ico at the root (if we really want, for IE lte10)
2. The `apple-touch-icon` that we can cache with a long TTL (all modern(ish) browsers will pick this one)

And that is pretty much it?

I would also like to, in the very near future, add an SVG for modern browsers with embedded CSS to respond to light and dark modes. But that is separate from this PR.

peterbe replied:
> an apple-touch-icon in the link tag

Both? `<link rel="apple-touch-icon" href="icon.png">` AND `<link rel="icon" href="icon.png">` where `icon.png` is the URL (with a hash in it). 

There's just something "stinky" about `apple-touch-icon` in that it's called "apple". It's a great company with great products but it's a company, not a generic web "standard" thing.

myakura replied:
>     1. favicon.ico at the root (if we really want, for IE lte10)
> 
>     2. The `apple-touch-icon` that we can cache with a long TTL (all modern(ish) browsers will pick this one)
> 
> 
> And that is pretty much it?

the `apple-touch-icon` is for mobile browsers and with that rel value browsers wont pick it up and show it in the tab. we need `<link rel="icon" href="favicon.png">`.

so we need three files:

1. /favicon.ico at root (an ICO file, for legacy browsers)
2. `<link rel="icon" href="favicon-48.png">` (a PNG file, for modern browsers)
3. `<link rel="apple-touch-icon" href="icon-180.png">` (a PNG file, for mobile browsers)

and two link tags for PNG files.

```html
<link rel="icon" href="favicon-48x48.png">
<link rel="apple-touch-icon" href="icon-180x180.png">
```

> I would also like to, in the very near future, add an SVG for modern browsers with embedded CSS to respond to light and dark modes. But that is separate from this PR.

Cool!

myakura reviewed (commented):
I suggest

schalkneethling commented on client/public/index.html:
```diff
@@ -11,31 +11,28 @@
+      sizes="32x32"
+      href="%PUBLIC_URL%/favicon-32x32.png"
     />
-
-    <!-- first- and second-generation iPad: -->
     <link
-      rel="apple-touch-icon-precomposed"
-      sizes="72x72"
-      href="%PUBLIC_URL%/favicon72.png"
+      rel="icon"
+      type="image/png"
+      sizes="16x16"
+      href="%PUBLIC_URL%/favicon-16x16.png"
```
I think for our user audience that makes sense.

peterbe commented:
@myakura Good now?

myakura commented:
@peterbe looks good!
